### PR TITLE
vscode for lsp4jakarta consumption

### PR DESCRIPTION
**Pull request by dshimo**
- add build to gulp
- add jakartajdt to java
- exec jakarta ls and bind requests
- init server mode

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -9,39 +9,85 @@
...
+        toggleItem(window.activeTextEditor, item);
+    });
+    /**, error => {;
+        console.log("Jakarta language client was not ready. Did not initialize");
+        console.log(error);
+
+        item.text = "Jakarta LS $(thumbsdown)";
+        item.tooltip = "Language Server for Jakarta failed to start";
+    });
+    **/
```
Comment (kathrynkodama) [inline src/extension.ts:83]: Looks like we can remove this code

File: src/extension.ts [hunk 2]
```diff
@@ -124,10 +170,10 @@ export function registerFileWatcher(projectProvider: ProjectProvider): void {
 	});
 }
 
-function startupLanguageServer(context: ExtensionContext) {
+function startupLibertyLanguageServer(context: ExtensionContext) {
     //Start up Liberty Language Server
-    const languageServerPath = context.asAbsolutePath(path.join("jars","liberty-langserver-1.0-SNAPSHOT-jar-with-dependencies.jar"));
-    const debugArgs = "-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=localhost:1074";
+    const languageServerPath = context.asAbsolutePath(path.join("jars", LIBERTY_LS_JAR));
+    // const debugArgs = "-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=localhost:1074";
```
Comment (kathrynkodama) [inline src/extension.ts:176]: Add a comment if this is needed for debugging

File: src/definitions/lsp4jakartaLSRequestNames.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+// Jakarta Language API
+export const JAVA_HOVER_REQUEST = "jakarta/java/hover";
```
Comment (kathrynkodama) [inline src/definitions/lsp4jakartaLSRequestNames.ts:2]: New class needs a license header

File: src/util/javaServerMode.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    LIGHTWEIGHT = "LightWeight",
+    HYBRID = "Hybrid",
+}
+
+/**
+ * Waits for the java language server to launch in standard mode
+ * Before activating Tools for MicroProfile.
+ * If java ls was started in lightweight mode, It will prompt user to switch
+ */
+export async function waitForStandardMode(api: JavaExtensionAPI): Promise<void> {
```
Comment (kathrynkodama) [inline src/util/javaServerMode.ts:17]: license header needed. If part of this code is re-used from another open-source project we should credit the original license as well

File: src/extension.ts [hunk 5]
```diff
@@ -9,39 +9,85 @@
...
+     * Before activating Tools for MicroProfile.
+     * If java ls was started in lightweight mode, It will prompt user to switch
+     */
+    const api: JavaExtensionAPI = await getJavaExtensionAPI();
+    await waitForStandardMode(api);
+
+    const item = window.createStatusBarItem(StatusBarAlignment.Right, Number.MIN_VALUE);
     // item.name = "Liberty Language Server";
     item.text = "Liberty LS";
     item.tooltip = "Language Server for Liberty is starting...";
```
Comment (travis1111) [inline src/extension.ts:44]: Translation

File: src/util/javaServerMode.ts [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * 
+ * Referenced:
+ * https://github.com/redhat-developer/vscode-microprofile/blob/master/src/util/javaServerMode.ts
+ */
```
Comment (kathrynkodama) [inline src/util/javaServerMode.ts:18]: Minor suggestion to move this to the top class (not as part of the license header)

**Review by kathrynkodama — APPROVED** [review]
LGTM pending the changes from @travis1111 and @TrevCraw thanks!

File: README.md [hunk 7]
```diff
@@ -69,11 +69,13 @@ Our [CONTRIBUTING](CONTRIBUTING.md) document contains details for submitting pul
 To build the extension locally:
 
 1. `git clone https://github.com/OpenLiberty/liberty-tools-vscode`
-2. `cd liberty-tools-vscode`
-3. Run `npm install`
-4. Run `npm run build`
-5. Run `npm run compile`
-4. Run the extension in Debug and Run mode by selecting `Run Extension` or `F5`
+2. `git clone https://github.com/eclipse/lsp4jakarta.git`
```
Comment (TrevCraw) [inline README.md:72]:
```suggestion
2. `git clone https://github.com/eclipse/lsp4jakarta.git` - Make sure the `lsp4jakarta` and `liberty-tools-vscode` projects are located in the same directory.
```

**Review by TrevCraw — APPROVED** [review]

**Pull request merged**